### PR TITLE
PUBAPI-1371 List volumes should not return deleted or failed instances

### PR DESCRIPTION
**Pull request by joshwilsdon**

### CODE REVIEW

**Comment by joshwilsdon** [comment]
still needs tests before it's worth reviewing

File: lib/endpoints/volumes.js [hunk 1]
```diff
@@ -110,6 +110,11 @@ function listVolumes(req, res, next) {
 
     if (req.params.predicate) {
         listVolumesParams.predicate = req.params.predicate;
+    } else if (!req.params.state) {
+        // PUBAPI-1371: When we have no state and no predicate, we should
+        // filter out deleted and failed volumes.
+        listVolumesParams.predicate
+            = '{"and":[{"ne":["state","deleted"]},{"ne":["state","failed"]}]}';
```
Comment (misterdjules) [inline lib/endpoints/volumes.js]: nit: using `JSON.stringify(predicateObject)` and defining `predicateObject` as a JS object and not a string might be easier to write/read/change, but that's really fine if you like the current form better.

**Comment by misterdjules** [comment]
Also, FWIW, just created https://smartos.org/bugview/PUBAPI-1397.

File: lib/endpoints/volumes.js (hunk 1 above)
Comment (joshwilsdon) [inline lib/endpoints/volumes.js]: good point, I've updated this now.

**Comment by misterdjules** [comment]
Still looks good to me after the latest changes.

**Pull request closed without merging**